codalab/apps/api/views/competition_views.py

```diff
@@ -685,3 +686,90 @@ def get(self, request, *args, **kwargs):
 class DefaultContentViewSet(viewsets.ModelViewSet):
     queryset = webmodels.DefaultContentItem.objects.all()
     serializer_class = serializers.DefaultContentSerial
+
+
+class SubmissionScoreView(views.APIView):
+    """
+    Provides a way to grab scores given a specific PK and the owner is the one making the request
+    """
+    def get(self, request, *args, **kwargs):
+        submission_id = self.kwargs.get('submission_id')
+        try:
+            sub = CompetitionSubmission.objects.get(pk=submission_id)
+            if not sub.participant.user == self.request.user:
+                temp_data = {
+                    'error': 'Not authorized!'
+                }
+                response = Response(temp_data, status=status.HTTP_403_FORBIDDEN)
+                return response
+            try:
+                scores = sub.phase.scores()
+                headers = list(sorted(scores[0]['headers'], key=lambda x: x.get('ordering')))
+                default_score_key = headers[0]['key']
+
+                for group in scores:
+                    for _, scoredata in group['scores']:
+                        try:
+                            default_score = next(val for val in scoredata['values'] if val['name'] == default_score_key)
+                            if int(scoredata['id']) == int(submission_id):
+                                temp_data = {
+                                    'score': default_score['val'],
+                                    'status': sub.status.codename
+                                }
+                                response = Response(temp_data, status=status.HTTP_200_OK)
+                                return response
+                        except (KeyError, StopIteration):
+                            pass
+            except (KeyError, IndexError):
+                pass
+        except ObjectDoesNotExist:
+            temp_data = {
+                'error': 'Object not found!'
+            }
+            response = Response(temp_data, status=status.HTTP_404_NOT_FOUND)
+            return response
+        temp_data = {
+            'error': 'Submission is not on leaderboard or is not accessible!'
+        }
+        response = Response(temp_data, status=status.HTTP_404_NOT_FOUND)
+        return response
+
+
+class AddChagradeBotView(views.APIView):
+    """
+    Provides a way to add a dummy chagrade bot user to competitions to make submissions from chagrade
+    """
+    def post(self, request, *args, **kwargs):
+        competition_id = self.kwargs.get('competition_id')
+        try:
+            comp = Competition.objects.get(pk=competition_id)
+            if not comp.creator == self.request.user and self.request.user not in comp.admins.all():
+                temp_data = {
```

Comment:
Can just raise permission denied?

Reply:
Django shortcut responses like Http404() don't work here. I can try with PermissionDenied but I think it's the same case.

Reply:
Raise permission denied should work?